drv/ignition-api/src/lib.rs

```diff
@@ -66,6 +61,24 @@ bitfield! {
     pub polarity_inverted, _: 2;
 }
 
+bitfield! {
+    #[derive(Copy, Clone, Debug, PartialEq, Eq, From, FromBytes, AsBytes)]
+    #[repr(C)]
+    pub struct SystemFaults(u8);
+    pub power_a3, _: 0;
+    pub power_a2, _: 1;
+    pub reserved1, _: 2;
+    pub reserved2, _: 3;
+    pub sp, _: 4;
+    pub rot, _: 5;
+}
+
+impl SystemFaults {
+    pub fn count(&self) -> usize {
+        self.0 as usize
+    }
```

Comment:
I think this just returns the value of `self.power_a3`. I believe you want to count the ones in the bitfield.